### ISSUE TITLE
Appears to not capture output sent from brew to STDERR

### ISSUE THREAD

**Issue by haakonstorm** [robocopAlpha/brewlog#2]
I'm writing a small command line command cacher.
That is, a function cached_output() in zsh, runs whatever is given to it as parameters.

e.g:

```
brewlog archive
cached_output brewlog search abcprogram
```

will archive the brew log, then run 

```
brewlog search abcprogram 
```

..in a subshell. But afterwards, `brewlog tail` only lists the date and the command, no output.

**Comment by dchakro** [comment]
You're right in saying that it doesn't capture the "Error" of `brewlog search abcprogram` but I think it is because it is not returned from brew as STDERR. Because if you try to run any of these:
1.  `brewlog upgrade abcprogram` 
2. `brewlog switch abcprogram 1.1 `
3. `brewlog update abcprogram`  (this is totally wrong, but just for demo)
and  thenrun `brewlog tail` you will notice the errors being logged.

These "errors" from `brew search` seem to be an interesting case.
If I try to capture everything from the `brew search` command by running `brew search abcprogram &> tmp.txt ; cat tmp.txt` or `brew search abcprogram > tmp.txt 2>&1 ; cat tmp.txt`, I get nothing.

I will dig further and find a fix for it.
Thanks a lot for reporting this issue!!

**Comment by dchakro** [comment]
It seems that `brew search`is not giving the error message `No formula or cask found for "abcprogram".` as STDERR.
If you run this line: you should see the error message (if any) in red color.
`brew update abcprogram 2> >(while read line; do echo -e "\e[01;31m$line\e[0m" >&2; done)`

However, if you see the following line you don't see it in red color, i.e. it is not sending that message via STDERR.
`brew search abcprogram 2> >(while read line; do echo -e "\e[01;31m$line\e[0m" >&2; done)`

I opened [an issue with homebrew](https://github.com/Homebrew/brew/issues/8870). Let's see if we get any answers.

**Comment by haakonstorm** [comment]
Great! Did you check to see whether `brew search` in fact perhaps opens another, unrelated, perhaps a `named pipe`? I don't really know how to debug shell scripts, but perhaps spawning a `zsh` or `bash` shell in a debugging environment is possible?

There is a myriad possibilities, but if brew creates a named pipe, since it isn't running with superuser privileges, the named pipe wouldn't be mounted in `/var/run`. 
Perhaps `/run/user/<userid>/.` or `/tmp` or `/var/tmp`. Or perhaps not a named pipe at all.

You are capturing both stdout and stderr? (`fd 1` and `fd 2`?) 
Perhaps `brew` are doing something else entirely when they output text?
Something is very strange indeed!

**Comment by dchakro** [comment]
Seems to have been addressed promptly by @whoiswillma 🥇 in https://github.com/Homebrew/brew/pull/8925 .
We just have to wait for a new release of brew 👍🏼  :)

**Comment by dchakro** [comment]
Closing as it is resolved with [brew release 2.5.7](https://github.com/Homebrew/brew/releases/tag/2.5.7).

To address the issue run:
```sh
brewlog update
brewlog search abcprogram
brewlog tail
```
